src/main/java/pw/itr0/kaba/encrypt/PasswordBasedDecrypter.java

```diff
@@ -30,9 +52,26 @@
      * @param password decrypting password
      * @param secret   secret bytes (Base64 encoded string)
      * @return decrypted secret
-     * @throws GeneralSecurityException on decryption failure
+     * @throws InvalidKeyException       if the {@code password} does not satisfy password specification of using decrypting algorithm,
+     *                                   or decrypting algorithm using specified password length is not usable on current JDK
+     * @throws BadPaddingException       if the {@code secret} is not properly padded
+     * @throws IllegalBlockSizeException if the {@code secret} has wrong length
      */
-    default byte[] decrypt64(char[] password, String secret) throws GeneralSecurityException {
+    default byte[] decrypt64(char[] password, String secret)
+            throws InvalidKeyException, BadPaddingException, IllegalBlockSizeException {
         return this.decrypt(password, Base64.getDecoder().decode(secret));
     }
+
+    /**
+     * Wrap {@link InputStream} {@code in} to {@link CipherInputStream} decrypting with {@code password}.
+     *
+     * @param in       {@link InputStream} to wrap in {@link CipherInputStream}
+     * @param password decrypting password
+     * @return decrypting {@link InputStream}
+     * @throws InvalidKeyException if the {@code password} does not satisfy password specification of using decrypting algorithm,
+     *                             or decrypting algorithm using specified password length is not usable on current JDK
+     */
+    default InputStream wrap(InputStream in, char[] password) throws InvalidKeyException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid prefixing parameters by in, out or inOut. Uses Javadoc to document this behavior.](https://www.codacy.com/app/ryotan/kaba-vault/file/2671050499/issues/source?bid=3298589&fileBranchId=3332567#l74)